plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -348,24 +366,35 @@ private static DeltaLakeColumnMetadata mapColumn(TypeManager typeManager, JsonNo
         JsonNode typeNode = node.get("type");
         boolean nullable = node.get("nullable").asBoolean();
         Type columnType = buildType(typeManager, typeNode, false);
+        OptionalInt fieldId = OptionalInt.empty();
         String physicalName;
         Type physicalColumnType;
-        if (mappingMode == ColumnMappingMode.NAME) {
-            physicalName = node.get("metadata").get("delta.columnMapping.physicalName").asText();
-            verify(!isNullOrEmpty(physicalName), "physicalName is null or empty");
-            physicalColumnType = buildType(typeManager, typeNode, true);
-        }
-        else {
-            physicalName = fieldName;
-            physicalColumnType = columnType;
+        switch (mappingMode) {
```

Comment:
nit(optional): IDE suggests a Java 17 enhanced switch

Reply:
I avoided it intentionally because there're other `switch` that can use enhanced version in this class and I didn't want to mix up in class.